install_config/topics/configuring_for_gce_registry-proc.adoc

```diff
@@ -0,0 +1,289 @@
+----
+
+[NOTE]
+====
+A bucket's data is automatically encrypted using a Google-managed key by
+default. To specify a different key to encrypt the data, see the
+https://cloud.google.com/storage/docs/encryption/[Data Encryption Options]
+available in GCP.
+====
+
+== Configuring {product-title} registry for GCP
+
+.Procedure
+
+To configure the Ansible inventory for the registry to use a `Google Cloud
+Storage` (`GCS`) bucket:
+
+[source,yaml]
+----
+[OSEv3:vars]
+# GCP Provider Configuration
+openshift_hosted_registry_storage_provider=gcs
+openshift_hosted_registry_storage_kind=object
+openshift_hosted_registry_replicas=1 <1>
+openshift_hosted_registry_storage_gcs_bucket=<bucket_name> <2>
+openshift_hosted_registry_storage_gcs_keyfile=<bucket_keyfile> <3>
+openshift_hosted_registry_storage_gcs_rootdirectory=<registry_directory> <4>
+----
+<1> The number of replicas to configure.
+<2> The bucket name to be used as registry storage.
+<3> The path on the installer host where the bucket's keyfile is located if
+using a custom key file to encrypt the data.
+<4> Directory used to store the data. `/registry` by default
+
+For more information, see link:https://cloud.google.com/storage/docs/[Cloud Storage in the GCP documentation].
+
+== Manually configuring {product-title} registry for GCP
+
+.Procedure
+
+To use GCP object storage, edit the registry’s configuration file and mount to the registry pod.
+
+. Take a default YAML configuration file as a base and replace the `filesystem`
+entry in the storage section with the GCP entry such as below. The resulting
+storage section may look like this:
++
+[source,yaml]
+----
+storage:
+  delete:
+    enabled: true
+  cache:
+    blobdescriptor: inmemory
+  gcs:
+    bucket: ocp-registry <1>
+    keyfile: mykeyfile <2>
+----
+<1>	Replace with the GCP bucket name.
+<2> A private service account key file in JSON format. If using the Google Application Default Credentials, do not specify the `keyfile` parameter.
++
+See the https://docs.docker.com/registry/storage-drivers/gcs/[Google Cloud Storage Driver documentation] for more information about storage driver configuration files.
+
+. Export the current *_config.yml_*:
++
+[source,bash]
+----
+$ oc get secret registry-config \
+    -o jsonpath='{.data.config\.yml}' -n default | base64 -d \
+  >> config.yml.old
+----
+
+. Edit the file to include the GCP parameters as explained before:
++
+[source,bash]
+----
+$ cp config.yml.old config.yml
+$ vim config.yml
+----
+
+. Delete the registry-config secret:
++
+[source,bash]
+----
+$ oc delete secret registry-config -n default
+----
+
+. Recreate the configmap to reference the updated configuration file:
++
+[source,bash]
+----
+$ oc create secret generic registry-config \
+    --from-file=config.yml -n default
+----
+
+. Redeploy the registry to read the updated configuration:
++
+[source,bash]
+----
+$ oc rollout latest docker-registry -n default
+----
+
+=== Verify the registry is using GCP object storage
```

Comment:
I tried to recreate this section as a procedure, but I think I'm off the mark. Are all the steps here necessary? If we're giving the reader options, can we just pick the one most likely to work for them? Is there something I'm missing?

Reply:
All the steps are required. The registry configuration is stored in a secret so in order to modify it it is required to export it, modify it and create a new one (and redeploy it) as secrets cannot be edited directly as they are base64 encoded.